core/src/jsMain/kotlin/Peripheral.kt

```diff
@@ -280,10 +337,9 @@ public class JsPeripheral internal constructor(
     private var isDisconnectedListenerRegistered = false
     private val disconnectedListener: (JsEvent) -> Unit = {
         logger.debug { message = GATT_SERVER_DISCONNECTED }
-        _state.value = State.Disconnected()
-        unregisterDisconnectedListener()
-        observationListeners.clear()
-        connectJob = null
+        scope.async {
+            disconnectSync(cancelConnectJob = true)
+        }
```

Comment:
Why does a disconnect operation need to be spun up asynchronously from this callback?

Reply:
Testing has shown that it is problematic to perform any `gatt` operations from within the callback. Doing so can call the system to become unresponsive and the `gatt` promises tend to go into limbo and neither reject nor resolve.

This behavior is not documented, but it easily replicated. See [comment 28 on this report](https://bugs.chromium.org/p/chromium/issues/detail?id=621901#c28) for example.